_includes/header.html

```diff
@@ -2,7 +2,13 @@
 
   <div class="wrapper">
 
-    <a class="site-title" href="{{ site.baseurl }}/">{{ site.title | escape }}</a>
+    <a class="site-title" href="{{ site.baseurl }}/">
+      {% if site.logo %}
+        <img src="/assets/{{ site.logo }}" class="site-logo" width="150" height="auto">
```

Comment:
There is really nothing you can do.  I saw this situation coming when this feature was made and tried to make suggestions to ease it but those suggestions were not accepted so all you can do is keep on keeping on until it becomes a problem.